src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -0,0 +1,138 @@
+package qna.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import java.util.List;
+import java.util.Optional;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.transaction.annotation.Transactional;
+
+@DataJpaTest
+class AnswerRepositoryTest {
+
+    @Autowired
+    private AnswerRepository answerRepository;
+
+    @Transactional
+    @Test
+    @DisplayName("저장한 Answer 의 id가 null 이 아닌지, 저장 전/후의 데이터 값이 같은지 확인")
+    void create() {
+        Answer answer = AnswerTest.A1;
+        Answer savedAnswer = answerRepository.save(answer);
+
+        assertAll(
+                () -> assertThat(savedAnswer.getId()).isNotNull(),
+                () -> assertThat(savedAnswer.getWriterId()).isEqualTo(answer.getWriterId()),
+                () -> assertThat(savedAnswer.getContents()).isEqualTo(answer.getContents()),
+                () -> assertThat(savedAnswer.getQuestionId()).isEqualTo(answer.getQuestionId())
+        );
+    }
+
+    @Transactional
+    @Test
+    @DisplayName("저장한 Answer 와 조회한 Answer 가 같은지 (동일성)확인")
+    void read() {
+        Answer answer = AnswerTest.A1;
+        Answer savedAnswer = answerRepository.save(answer);
+
+        Optional<Answer> findAnswer = answerRepository.findById(savedAnswer.getId());
+
+        assertAll(
+                () -> assertThat(findAnswer).isPresent(),
+                () -> assertThat(savedAnswer).isSameAs(findAnswer.get())
+        );
+    }
+
+    @Transactional
+    @Test
+    @DisplayName("저장한 Answer 의 Question 변경 후 조회한 Answer 의 Question 과 같은지 확인")
+    void update() {
+        Answer savedAnswer = answerRepository.save(AnswerTest.A1);
+        savedAnswer.toQuestion(QuestionTest.Q2);
+
+        Optional<Answer> findAnswer = answerRepository.findById(savedAnswer.getId());
+
+        Assertions.assertThat(savedAnswer.getQuestionId()).isEqualTo(findAnswer.get().getQuestionId());
+    }
+
+    @Transactional
+    @Test
+    @DisplayName("저장한 Answer 삭제 후 조회 시 Answer 가 없는지 확인")
+    void delete() {
+        Answer savedAnswer = answerRepository.save(AnswerTest.A1);
+        answerRepository.delete(savedAnswer);
+
+        Optional<Answer> findAnswer = answerRepository.findById(savedAnswer.getId());
+
+        Assertions.assertThat(findAnswer).isNotPresent();
+    }
+
+    @Transactional
+    @Test
+    @DisplayName("Answer 의 deleted 가 false 일 때 findByIdAndDeletedFalse 로 조회된 Answer 가 있는지 확인")
+    void findByIdAndDeletedFalse() {
+        Answer savedAnswer = answerRepository.save(AnswerTest.A1);
+        savedAnswer.setDeleted(false);
```

Comment:
```suggestion
        savedAnswer.deleted();
```

`deleted` 삭제 한다는것인데, `true` or `false`를 전달 받는게 맞을지 고민이 되네요 :thinking: 
삭제한다면, 이미 데이터는 정해져 있는게 아닐까요?

Reply:
[Getter, Setter 지양하기](https://velog.io/@sezeom/Getter-Setter-%EC%A7%80%EC%96%91%ED%95%98%EA%B8%B0)

참고해주세요~

Reply:
2단계 미션에서 리뷰해주신 내용 반영해보도록 하겠습니다!!